From 798c2eb543c44c0bc57d6337e2e4374fff09a86b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 31 Jan 2018 19:52:40 +0800
Subject: [PATCH 3303/5242] MLK-17473-7 drm/imx: dec400d: avoid shadow trigger
 when bypass dec400d

commit  8882c674a8c9fe3f937094c2ab8e8eb4d4bf653f from
https://source.codeaurora.org/external/imx/linux-imx.git

Do not really do shadow regiters trigger in'dcss_dec400d_shadow_trig()'
when dec400d is bypassed, since in 'dcss_dec400d_bypass()', the shadow
registers have already been triggerd.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dec400d.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-dec400d.c b/drivers/gpu/imx/dcss/dcss-dec400d.c
index 0bcd12b..874fd5f 100644
--- a/drivers/gpu/imx/dcss/dcss-dec400d.c
+++ b/drivers/gpu/imx/dcss/dcss-dec400d.c
@@ -134,6 +134,10 @@ void dcss_dec400d_shadow_trig(struct dcss_soc *dcss)
 	uint32_t control;
 	struct dcss_dec400d_priv *dec400d = dcss->dec400d_priv;
 
+	/* do nothing */
+	if (dec400d->bypass == true)
+		return;
+
 	control = dcss_readl(dec400d->dec400d_reg + DEC400D_CONTROL);
 
 	/* Trigger shadow registers */
-- 
1.7.9.5

